src/LEPTON/fix_epot_lepton.cpp

```diff
@@ -0,0 +1,324 @@
+  expr = LeptonUtils::condense(arg[3]);
+  try {
+    auto parsed = Lepton::Parser::parse(LeptonUtils::substitute(expr, lmp));
+    auto phi = parsed.createCompiledExpression();
+  } catch (std::exception &e) {
+    error->all(FLERR, e.what());
+  }
+
+  force_flag = 0;
+  fsum[0] = fsum[1] = fsum[2] = fsum[3] = 0.0;
+}
+
+/* ---------------------------------------------------------------------- */
+
+FixEpotLepton::~FixEpotLepton()
+{
+  delete[] idregion;
+}
+
+/* ---------------------------------------------------------------------- */
+
+int FixEpotLepton::setmask()
+{
+  int mask = 0;
+  mask |= POST_FORCE;
+  mask |= POST_FORCE_RESPA;
+  mask |= MIN_POST_FORCE;
+  return mask;
+}
+
+/* ---------------------------------------------------------------------- */
+
+void FixEpotLepton::init()
+{
+  if (!atom->q_flag && !atom->mu_flag)
+    error->all(FLERR, "Fix {} requires atom attribute q or mu", style);
+  if (atom->mu_flag && !atom->torque_flag)
+    error->all(FLERR, "Dipoles must be finite-sized to rotate", style);
+
+  // set index and check validity of region
+  if (idregion) {
+    region = domain->get_region_by_id(idregion);
+    if (!region) error->all(FLERR, "Region {} for fix {} does not exist", idregion, style);
+  }
+
+  if (utils::strmatch(update->integrate_style, "^respa")) {
+    ilevel_respa = (dynamic_cast<Respa *>(update->integrate))->nlevels - 1;
+    if (respa_level >= 0) ilevel_respa = MIN(respa_level, ilevel_respa);
+  }
+}
+
+/* ---------------------------------------------------------------------- */
+
+void FixEpotLepton::setup(int vflag)
+{
+  if (utils::strmatch(update->integrate_style, "^respa")) {
+    auto respa = dynamic_cast<Respa *>(update->integrate);
+    respa->copy_flevel_f(ilevel_respa);
+    post_force_respa(vflag, ilevel_respa, 0);
+    respa->copy_f_flevel(ilevel_respa);
+  } else {
+    post_force(vflag);
+  }
+}
+
+/* ---------------------------------------------------------------------- */
+
+void FixEpotLepton::min_setup(int vflag)
+{
+  post_force(vflag);
+}
+
+/* ----------------------------------------------------------------------
+  Apply F = qE,
+        F = (mu . D) E,
+        T = mu x E
+------------------------------------------------------------------------- */
+
+void FixEpotLepton::post_force(int vflag)
+{
+  double **f = atom->f;
+  double **x = atom->x;
+  double *q = nullptr;
+  double **mu = nullptr, **t = nullptr;
+  int *mask = atom->mask;
+  imageint *image = atom->image;
+  int nlocal = atom->nlocal;
+
+  const double qe2f = force->qe2f;
+  const double qqr2e = force->qqr2e;
+
+  auto parsed = Lepton::Parser::parse(LeptonUtils::substitute(expr, lmp)).optimize();
+  Lepton::CompiledExpression phi, dphi_x, dphi_y, dphi_z, dphi_xx, dphi_xy, dphi_xz, dphi_yy, dphi_yz, dphi_zz; 
+
+  dphi_x = parsed.differentiate("x").createCompiledExpression();
+  dphi_y = parsed.differentiate("y").createCompiledExpression();
+  dphi_z = parsed.differentiate("z").createCompiledExpression();
```

Comment:
I'm genuinely not sure if maybe modern computers need less hand-holding -- but I feel like these CompiledExpressions should be defined in a header, allocated in an `allocate` function, and deallocated in the fix destructor, instead of being dynamically allocated and destroyed every step.

Also, since there are only ever 10 possible expressions (energy, 3 force, 6 torque), the code might actually be _more_ readable without the vectors and arrays.